clients/client-python/tests/integration/hdfs_container.py

```diff
@@ -0,0 +1,136 @@
+This software is licensed under the Apache License version 2.
+"""
+
+import asyncio
+import logging
+import time
+
+import docker
+from docker import types as tp
+from docker.errors import NotFound
+
+from gravitino.exceptions.gravitino_runtime_exception import GravitinoRuntimeException
+
+logger = logging.getLogger(__name__)
+
+
+async def check_hdfs_status(hive_container):
+    retry_limit = 15
+    for _ in range(retry_limit):
+        try:
+            command_and_args = ["bash", "/tmp/check-status.sh"]
+            exec_result = hive_container.exec_run(command_and_args)
+            if exec_result.exit_code != 0:
+                message = (
+                    f"Command {command_and_args} exited with {exec_result.exit_code}"
+                )
+                logger.warning(message)
+                logger.warning("output: %s", exec_result.output)
+                output_status_command = ["hdfs", "dfsadmin", "-report"]
+                exec_result = hive_container.exec_run(output_status_command)
+                logger.info("HDFS report, output: %s", exec_result.output)
+            else:
+                logger.info("HDFS startup success!")
+                return True
+        except Exception as e:
+            logger.error("While checking HDFS container status happen exception: %s", e)
+        time.sleep(10)
+    return False
+
+
+async def check_hive_container_status(hive_container):
+    timeout_sec = 150
+    try:
+        result = await asyncio.wait_for(
+            check_hdfs_status(hive_container), timeout=timeout_sec
+        )
+        assert result is True, "HDFS container startup failed!"
+    except asyncio.TimeoutError as e:
+        raise GravitinoRuntimeException(
+            "Timeout occurred while waiting for checking HDFS container status."
+        ) from e
+
+
+class HDFSContainer:
+    _docker_client = None
+    _container = None
+    _network = None
+    _ip = ""
+    _network_name = "python-net"
+    _container_name = "python-hdfs"
+
+    def __init__(self):
+        self._docker_client = docker.from_env()
+        self._create_networks()
+        try:
+            container = self._docker_client.containers.get(self._container_name)
+            if container is not None:
+                if container.status == "running":
+                    container.kill()
+                container.remove()
+        except NotFound:
+            logger.warning("Cannot find hdfs container in docker env, skip to remove.")
+        self._container = self._docker_client.containers.run(
+            image="datastrato/gravitino-ci-hive:0.1.12",
+            name=self._container_name,
+            detach=True,
+            environment={"HADOOP_USER_NAME": "datastrato"},
+            network=self._network_name,
+        )
+        asyncio.run(check_hive_container_status(self._container))
+
+        self._fetch_ip()
+
+    def _create_networks(self):
+        pool_config = tp.IPAMPool(subnet="10.20.31.16/28")
```

Comment:
Where this subnet comes from?

Reply:
Copy from https://github.com/datastrato/gravitino/blob/main/integration-test/trino-it/docker-compose.yaml#L105.